xarray/tests/test_dataset.py

```diff
@@ -5654,6 +5654,29 @@ def test_coarsen_coords_cftime():
     np.testing.assert_array_equal(actual.time, expected_times)
 
 
+def test_coarsen_keep_attrs():
+    _attrs = {"units": "test", "long_name": "testing"}
+
+    var1 = np.linspace(10, 15, 100)
+    var2 = np.linspace(5, 10, 100)
+    coords = np.linspace(1, 10, 100)
+
+    ds = Dataset(
+        data_vars={"var1": ("coord", var1), "var2": ("coord", var2)},
+        coords={"coord": coords},
```

Comment:
FYI you can just pass the `attrs` here rather than assigning them below; i.e. `attrs=_attrs`. Then you know they're consistent with those you're testing against in the assertion